edx_sga/static/js/src/edx_sga.js

```diff
@@ -314,10 +321,82 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $(self).addClass("disabled");
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(element).find('.task-message').hide();
+                      } else {
+                        $(element).find('.task-message').show();
+                        $(element).find('.task-message').html(comeBackMessage());
+                        $(element).find('.task-message').removeClass("ready-msg");
+                        $(element).find('.task-message').addClass("preparing-msg");
+                      }
+                      $(self).removeClass("disabled");
+                    }
+                  );
+                });
+                
+                pollUntilSuccess(downloadSubmissionsStatusUrl, checkResponse, 10000, 400).then(function(res) {
```

Comment:
We only want to start this polling _after_ the user has clicked "Download". We do not need to start checking for the submissions file until after the user actually requests one. This code starts the polling as soon as the page loads.